homeassistant/components/hive/water_heater.py

```diff
@@ -75,6 +75,14 @@ class HiveWaterHeater(HiveEntity, WaterHeaterEntity):
     _attr_temperature_unit = UnitOfTemperature.CELSIUS
     _attr_operation_list = SUPPORT_WATER_HEATER
 
+    @property
+    def extra_state_attributes(self):
+        """Return the additional state attributes."""
+        attributes = {}
+        attributes['operation_state'] = self._attr_current_state
```

Comment:
You appear to be accessing the `self._attr_current_state` before it is initialised so this would probably raise an error.  Have you been able to test this change in a dev environment?

By setting this as a `extra_state_attributes` how are you expecting users to see it on the UI? It's possible but requires some extra work. It would be better to create a `binary_sensor` to display this value automatically.

Reply:
Yes, I've tested this and it appears to work without any errors.

I agree that there is additional work required to display this attribute in the UI; but I see this change as the first step to facilitate this.  I'm currently using the new `current_state` attribute in an automation, which meets my requirements (_BTW, it was me that raised the original issue and, after waiting over a year of waiting for a resolution, I eventually decided to take a look myself - this the solution that I came up with_).

Reply:
The `_attr_` pattern is described [here](https://developers.home-assistant.io/docs/core/entity?_highlight=_attr#entity-class-or-instance-attributes).  They're called "shorthand attributes" for entities.  You have named your variable to look like a shorthand attribute on the base entity but it isn't one.  You should just give it an ordinary name without the prefix.

Also because you are not initialising it in `__init__` it doesn't exist prior to `async_update` being called so if the `extra_state_attributes` property is accessed prior to the update it will raise an error. I don't know the ins and outs of how the data refresh is handled in this integration so `async_update` is probably being called right at the start which is avoiding the error. None the less it is bad practise to have code that accesses variables that could be unbound and raise errors.

Reply:
Thanks for the link to "shorthand attributes" documentation.  I now understand the significance of the `_attr_` prefix.

I've updated the code, as requested.